File: server/src/main/java/de/ingrid/igeserver/utils/AuthUtilsMock.java
package de.ingrid.igeserver.utils;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;

import java.security.Principal;

@Service
@Profile("dev")
public class AuthUtilsMock implements AuthUtils {

    @Value("${dev.keycloak.enabled:true}")
    boolean isKeycloakEnabled;

    @Value("${dev.user.login:}")
    String mockedLogin;

    @SuppressWarnings("unused")
    public String getUsernameFromPrincipal(Principal principal) {
        // return a user for development when security is switched off
        return mockedLogin;
    }

}

File: server/src/main/java/de/ingrid/igeserver/utils/AuthUtilsMock.kt
package de.ingrid.igeserver.utils

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Service
import java.security.Principal

@Service
@Profile("dev")
class AuthUtilsMock : AuthUtils {

    @Value("\${dev.user.login:}")
    lateinit var mockedLogin: String

    override fun getUsernameFromPrincipal(principal: Principal?): String {
        // return a user for development when security is switched off
        return mockedLogin
    }

}